lib/array_equals.rb

```diff
@@ -1,5 +1,18 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
+
 def array_equals(array1, array2)
-  raise NotImplementedError
+  if array1 == nil || array2 == nil
+    return array1 == nil && array2 == nil
```

Comment:
I'd recommend returning `true` here in favor of `return array1 == nil && array2 == nil` for readability and maintenance.